cmd/dep/ensure.go

```diff
@@ -779,13 +786,47 @@ func getProjectConstraint(arg string, sm gps.SourceManager) (gps.ProjectConstrai
 	}
 
 	pi := gps.ProjectIdentifier{ProjectRoot: pr, Source: source}
+
+	// check to see whether our versionStr contains what looks to be a semver
+	// tag -- this is so we can make sure it's a full version and not a range
+	if len(versionStr) > 0 {
```

Comment:
Why not:
```go
if versionStr != "" {
```

Reply:
Off the top of my head I can only assume it's because I read the logic as `if there is data` instead of `if this is not empty`. This is usually the way I write this.